### PR TITLE
feat: Stabilize Deno.hostname() API

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/15927

### CODE REVIEW

File: runtime/ops/os.rs [hunk 1]
```diff
@@ -151,7 +151,6 @@ fn op_loadavg(state: &mut OpState) -> Result<(f64, f64, f64), AnyError> {
 
 #[op]
 fn op_hostname(state: &mut OpState) -> Result<String, AnyError> {
-  super::check_unstable(state, "Deno.hostname");
   state.borrow_mut::<Permissions>().env.check_all()?;
```
Comment (cmoog) [inline runtime/ops/os.rs:154]: It seems sub-optimal that this requires granting permission to *all* environment variables. If some dependency needs to read the hostname, now I need to expose my entire env.

This issue also exists for these other unstable APIs.
- loadavg
- osRelease
- networkInterfaces
- systemMemoryInfo
- getGid
- getUid

Any thoughts on introducing a new permissions flag `--allow-sys-info` that accepts hard coded keys to expose each of these features individually?

Comment (kt3k) [reply runtime/ops/os.rs:154]: I like the idea. I created an independent issue for it https://github.com/denoland/deno/issues/15986

**Comment by bartlomieju** [comment]
I will merge this PR after https://github.com/denoland/deno/pull/16028 is merged.

**Review by dsherret — APPROVED** [review]
LGTM

**Review by lucacasonato — APPROVED** [review]

**Pull request merged**